ddl/db_integration_test.go

```diff
@@ -2031,6 +2031,23 @@ func TestDefaultValueIsString(t *testing.T) {
 	require.Equal(t, "1", tbl.Meta().Columns[0].DefaultValue)
 }
 
+func TestDefaultColumnWithRand(t *testing.T) {
+	// Related issue: https://github.com/pingcap/tidb/issues/10377
+	store, clean := testkit.CreateMockStoreWithSchemaLease(t, testLease)
+	defer clean()
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+
+	tk.MustExec("drop table if exists t, t1, t2")
+	tk.MustExec("create table t (c int(10) default (rand()))")
+	tk.MustExec("create table t1 (c int default (rand()), c1 double default (rand()))")
+	tk.MustExec("create table t2 (c int default (rand(1)), c1 double default (rand(1)))")
+	tk.MustExec("alter table t add column c1 int(10) default (rand(2))")
+	tk.MustExec("insert into t(c) values (1),(2),(3)")
+	tk.MustExec("insert into t1(c) values (1),(2),(3)")
+	tk.MustExec("insert into t2(c) values (1),(2),(3)")
```

Comment:
do we have a way to test the inserted value

Reply:
Yes, I've add the checks for the inserted values. PTAL